### PR TITLE
Add bases2fastq

### PR DESCRIPTION
**Pull request by DriesSchaumont**
## Description

Add bases2fastq

## Issue ticket number

~Closes #xxxx~

## Checklist before requesting a review

- [x] I have performed a self-review of my code

- [x] Conforms to the [Contributing guidelines](https://github.com/viash-hub/base/blob/main/CONTRIBUTING.md)

- [x] Proposed changes are described in the [CHANGELOG.md](https://github.com/viash-hub/base/blob/main/CHANGELOG.md)

- [x] I have tested my code with `viash ns test --parallel -q <name or namespace>`

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Documentation
  - [ ] Bug fixes

### CODE REVIEW

File: src/bases2fastq/config.vsh.yaml [hunk 1]
```diff
@@ -0,0 +1,204 @@
+name: bases2fastq
+description: |
+  Bases2Fastq demultiplexes sequencing data generated by Element Biosciences instruments and converts base calls into FASTQ files.
+keywords: ["demultiplex", "fastq"]
```
Comment (tverbeiren) [inline src/bases2fastq/config.vsh.yaml:4]: + `demux`, `Element Biosciences`

File: src/bases2fastq/config.vsh.yaml [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+      - name: --r2_cycles
+        type: integer
+        min: 1
+        description: |
+          Run parameters override, R2 cycles.
+      - name: "--split_lanes"
+        type: boolean_true
+        description: |
+          Split FASTQ files by lane.
+      # - name: "--skip_qc_report"
```
Comment (tverbeiren) [inline src/bases2fastq/config.vsh.yaml:161]: Remove?

File: src/bases2fastq/config.vsh.yaml [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        type: boolean_true
+        description: |
+          In strict mode any invalid or missing input file will terminate execution 
+          (overrides no_error_on_invalid and sets --error_on_missing)
+  
+        # --help, -h                      Display this usage statement              
+        # --input-remote, NAME            Rclone remote name for remote ANALYSIS_DIRECTORY
+        # --num-threads, -p NUMBER        Number of threads (default 1)
+        # --output-remote, NAME           Rclone remote name for remote OUTPUT_DIRECTORY
+        # --settings SELECTION            Run manifest settings override. This option may be specified multiple times.
```
Comment (tverbeiren) [inline src/bases2fastq/config.vsh.yaml:175]: Not relevant? Or covered by the other arguments?

File: src/bases2fastq/test.sh [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+# can differ based on the arguments provided to bases2fastq
+
+# |-- 20230404-Bases2Fastq-Sim_QC.html
+# |-- IndexAssignment.csv
+# |-- Metrics.csv
+# |-- RunManifest.csv
+# |-- RunManifest.json
+# |-- RunParameters.json
+# |-- RunStats.json
+# |-- Samples
```
Comment (tverbeiren) [inline src/bases2fastq/test.sh:36]: This seems incompatible with how the output of other demultiplexers is structured. This makes sense here, but maybe we could make a note of that in the description of the output arguments?

File: src/bases2fastq/config.vsh.yaml (hunk 2 above)
Comment (DriesSchaumont) [reply src/bases2fastq/config.vsh.yaml:161]: Done!

File: src/bases2fastq/config.vsh.yaml (hunk 3 above)
Comment (DriesSchaumont) [reply src/bases2fastq/config.vsh.yaml:175]: Some are covered by other arguments, some by `meta` values. Some are just tricky/unclear to me what they do and/or are covered by nextflow.

File: src/bases2fastq/test.sh (hunk 4 above)
Comment (DriesSchaumont) [reply src/bases2fastq/test.sh:36]: I made sure that this component has similar output arguments to `bclconvert`: it also had `logs`, `reports`, and `output_directory` arguments? I'm also not sure what you mean with 'other demultiplexers', each vendor has their own output structure I think.

File: src/bases2fastq/config.vsh.yaml (hunk 1 above)
Comment (DriesSchaumont) [reply src/bases2fastq/config.vsh.yaml:4]: Added

**Pull request merged**